src/compiler/types.ts

```diff
@@ -1329,6 +1330,7 @@ module ts {
 
         // Values for enum members have been computed, and any errors have been reported for them.
         EnumValuesComputed  = 0x00000080,
+        BlockScopedBindingInLoop = 0x00000100,

```

Comment:
To be honest I don't think that there is any useful difference between the two. But there is a _technical_ difference, as the loop body is a new scope nested inside the loop scope itself. It means that you can redeclare `let i` both in the loop initializer and the loop body. The initializer, loop increment and loop condition share one scope, the loop body is a nested scope.

Allen makes it very clear in his answer to this question: https://esdiscuss.org/topic/in-es6-do-for-loops-with-a-let-const-initializer-create-a-separate-scope